src/role.ts

```diff
@@ -19,6 +19,52 @@ export interface GithubConfiguration {
    */
   readonly provider: IGithubActionsIdentityProvider;
 
+  /**
+   * Repository owner (organization or username).
+   *
+   * Use `trustedRepositories` if you want to provide multiple repo configurations
+   *
+   * @example
+   * 'octo-org'
+   */
+  readonly owner?: string;
+
+  /**
+   * Repository name (slug) without the owner.
+   *
+   * Use `trustedRepositories` if you want to provide multiple repo configurations
+   *
+   * @example
+   * 'octo-repo'
+   */
+  readonly repo?: string;
+
+  /**
+   * Subject condition filter, appended after `repo:${owner}/${repo}:` string in IAM Role trust relationship.
+   *
+   * @default
+   * '*'
+   *
+   * You may use this value to only allow Github to assume the role on specific branches, tags, environments, pull requests etc.
+   * @example
+   * 'ref:refs/tags/v*'
+   * 'ref:refs/heads/demo-branch'
+   * 'pull_request'
+   * 'environment:Production'
+   *
+   * @see https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect#examples
+   */
+  readonly filter?: string;
+
+  /**
+   * Provide multiple trusted repositories allowed to assume this role.
+   *
+   * @default - required if top-level owner/repo/filter not set
+   */
+  readonly trustedRepositories?: TrustedRepository[];
+}
+
+export interface TrustedRepository {
```

Comment:
`TrustedRepository` takes the same top-level `owner`/`repo` and optionally `filter` props.